services/worker-manager/src/providers/aws.js

```diff
@@ -230,7 +230,56 @@ class AwsProvider extends Provider {
     return {expires: taskcluster.fromNow('96 hours')};
   }
 
-  async checkWorker({worker});
+  async checkWorker({worker}) {
+    this.seen[worker.workerPoolId] = this.seen[worker.workerPoolId] || 0;
+
+    let instanceStatuses = (await this.ec2.describeInstanceStatus({
+      InstanceIds: [worker.workerId.toString()],
+    })).data;
+
+    Promise.all(instanceStatuses.map(is => {
+      switch (is.InstanceState.Name) {
+        case 'pending':
+        case 'running':
+          this.seen[worker.workerPoolId] += 1;
+          return Promise.resolve();
+
+        case 'shutting-down':
+        case 'terminated':
```

Comment:
I think we'll also mark it as stopped if it is terminated

Reply:
these are the statuses that come aws. Or do you mean.. something else?

Reply:
Ah, just that down below you have something like `case 'stopped':` where you mark the worker state as `STOPPED`. We should do that when the aws status is `terminated` as well.

Reply:
yep, that's exactly how this switch is supposed to work

Reply:
Ohhhh, duh. I was not reading that correctly. I wonder if we should keep `shutting-down` and `stopping` in the running state so that we don't turn on new instances until they're entirely gone?

Reply:
Sounds good to me!